### PR TITLE
CA-102: feat: add AppShellBloc (tab selection/lazy-load state) and CurrentProjectNotifierImpl

### PR DESCRIPTION
**Pull request by ayanasamuel8**
### Summary
Introduces `AppShellBloc` to manage tab navigation state in the app shell, including lazy-loading optimization for tab content.

### Changes
- Add `AppShellBloc` with tab selection handling
- Track loaded tabs via `loadedTabIndexes` to support lazy initialization
- Skip redundant state emissions when selecting the current tab
- Add unit tests for initial state, tab transitions, and edge cases

### Testing
- [x] Unit tests added (`app_shell_bloc_test.dart`)
- [x] All tests passing

### CODE REVIEW

**Review by SimonDerejeW — APPROVED** [review]
Just a heads-up: tests that are located outside the `feature/` and `library/` directories are currently not included in our scripts or CI/CD pipeline for coverage reporting.

**Review by ripplearcgit — CHANGES_REQUESTED** [review]
The `AppShellBloc` and `CurrentProjectNotifierImpl` production code are solid — clean state design, correct broadcast controller usage, proper `Disposable` impl. Three things need addressing before this merges.

**Dependency chain:** This PR bases on `C-229-2-dashboard-refactor` (#204). Review that branch first.

**Review by SimonDerejeW — CHANGES_REQUESTED** [review]

File: test/features/project/units/presentation/current_project_notifier_impl_test.dart [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        final emittedIds = <String?>[];
+        final subscription = notifier.onCurrentProjectChanged.listen(
+          emittedIds.add,
+        );
+
+        notifier.setCurrentProjectId(newId);
+
+        // Allow stream to emit
+        await Future<void>.delayed(Duration.zero);
+
```
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply test/features/project/units/presentation/current_project_notifier_impl_test.dart]: You're totally right that the manual emittedIds array and await pumpEventQueue() pattern was messy, flaky, and misleading to read.

However, just a quick technical note: Dart's StreamController.broadcast() is actually asynchronous by default (it schedules events on the microtask queue), which is why the tests were failing without pumpEventQueue(). Changing the app's controller to sync: true isn't ideal because synchronous streams can block execution or cause StateErrors if events are nested.

Instead, I agreed with your point about cleaning it up, so I wiped out all the messy manual arrays and pumpEventQueues entirely. I refactored the tests to use Dart's native async stream matchers (expectLater(notifier.onCurrentProjectChanged, emits(newId))). This natively tests the async stream and completely removes the flakiness you correctly spotted!

**Review by SimonDerejeW — COMMENTED** [review]
I think there should be documentations for our events and states. I think the linter should have caught it. Please check it and add documentation for them.

**Review by ripplearcgit — CHANGES_REQUESTED** [review]
Two blocking items remain unresolved at HEAD.

---

🔴 **`TODO(integration-pr)` in `project_module.dart` has no YouTrack ticket link**

Ayana confirmed in the thread there is no ticket for the `CurrentProjectNotifier` → `ProjectDropdownBloc` integration wiring. A YouTrack task has been drafted at https://ripplearc.youtrack.cloud/newIssue?draftId=3-1480 — please publish it, get the CA-XXX ID, and update the TODO to the required format before merge:

```dart
// TODO: [CA-XXX] Wire CurrentProjectNotifier via ProjectDropdownBloc in AppShellPage.
// https://ripplearc.youtrack.cloud/issue/CA-XXX
```

---

🔴 **Missing `///` doc comments on all public BLoC members**

`AppShellBloc`, `AppShellEvent`, `AppShellTabSelected` (and its `index` field), `AppShellState` (and its fields + `copyWith`) all still have no `///` doc comments at HEAD. This was explicitly requested by Simon and remains unresolved.

**Review by ripplearcgit — COMMENTED** [review]
🔴 **`AppShellBloc` test uses unnecessary Modular DI — direct instantiation is required**

`AppShellBloc()` takes no constructor arguments. The `_AppShellBlocTestModule` and `Modular.init/destroy` lifecycle exist purely to call `AppShellBloc.new` — all of that overhead can and should be replaced with direct instantiation, which is the established pattern for dependency-free BLoCs.

File: test/app/shell/units/app_shell_bloc_test.dart [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+class _AppShellBlocTestModule extends Module {
+  @override
+  void binds(Injector i) {
+    i.add<AppShellBloc>(AppShellBloc.new);
+  }
+}
+
+void main() {
+  setUp(() {
```
[the remark this answers is not in the document]
Comment (ayanasamuel8) [reply test/app/shell/units/app_shell_bloc_test.dart]: :+1:

File: test/app/shell/units/app_shell_bloc_test.dart [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import 'package:bloc_test/bloc_test.dart';
```
[the remark this answers is not in the document]
Comment (ripplearcgit) [reply test/app/shell/units/app_shell_bloc_test.dart]: could you work @SimonDerejeW to add it first? 

Before that happens, this is blocking.

**Review by ripplearcgit — CHANGES_REQUESTED** [review]

**Review by ripplearcgit — APPROVED** [review]
Please use DI in the test

**Review by SimonDerejeW — APPROVED** [review]

**Comment by ayanasamuel8** [comment]
#runcheck
